tests/unit/dataactvalidator/test_b12_award_financial_1.py

```diff
@@ -0,0 +1,75 @@
+from tests.unit.dataactcore.factories.staging import AwardFinancialFactory
+from tests.unit.dataactvalidator.utils import number_of_errors, query_columns
+
+
+_FILE = 'b12_award_financial_1'
+
+
+def test_column_headers(database):
+    expected_subset = {'row_number', 'by_direct_reimbursable_fun'}
+    actual = set(query_columns(_FILE, database))
+    assert (actual & expected_subset) == expected_subset
+
+
+def test_success(database):
+    """ Test for USSGL 48XX & 49XX (except 487X & 497X) if any one is provided then
+    by_direct_reimbursable_fun is not empty """
+
+    af = AwardFinancialFactory()
+    assert number_of_errors(_FILE, database, models=[af]) == 0
+
+
+def test_failure(database):
+    """ Test for USSGL 48XX & 49XX (except 487X & 497X) if any one is provided then
+    by_direct_reimbursable_fun is empty """
+
+    af = AwardFinancialFactory(ussgl480100_undelivered_or_fyb=None, by_direct_reimbursable_fun=None)
+    assert number_of_errors(_FILE, database, models=[af]) == 1
```

Comment:
If I understand what's going on, the rule is triggered when `by_direct_reimbursable_fun` is null and _any_ of a set of other fields are not null. In each of these assertions, we've got all of those fields set except for one -- I'm wondering if it should be the other way around: all of the fields are _null_ except for one.